cloud/scope/vpc_cluster.go

```diff
@@ -486,9 +497,186 @@ func (s *VPCClusterScope) createVPC() (*vpcv1.VPC, error) {
+		requeue := true
+		if image.Status != nil && *image.Status == string(vpcv1.ImageStatusAvailableConst) {
+			requeue = false
+		}
+		s.SetResourceStatus(infrav1beta2.ResourceTypeCustomImage, &infrav1beta2.ResourceStatus{
+			ID:   *imageID,
+			Name: image.Name,
+			// Ready status will be invert of the need to requeue.
+			Ready: !requeue,
+		})
+		return requeue, nil
+	}
+
+	// No VPC Custom Image exists or was found.
+	// So, check if the Image spec was defined, as it contains all the data necessary to reconcile.
+	if s.IBMVPCCluster.Spec.Image == nil {
+		// If no Image spec was defined, we expect it is maintained externally and continue without reconciling. For example, using a Catalog Offering Custom Image, which may be in another account, which means it cannot be looked up, but can be used when creating Instances.
+		s.V(3).Info("No VPC Custom Image defined, skipping reconciliation")
+		return false, nil
+	}
+
+	// Create Custom Image.
+	s.Info("Creating a VPC Custom Image")
+	image, err := s.createCustomImage()
+	if err != nil {
+		return false, fmt.Errorf("error failure trying to create vpc custom image: %w", err)
+	} else if image == nil {
+		return false, fmt.Errorf("error no vpc custom image creation results")
+	}
+
+	s.Info("Successfully created VPC Custom Image")
+	s.SetResourceStatus(infrav1beta2.ResourceTypeCustomImage, &infrav1beta2.ResourceStatus{
+		ID:   *image.ID,
+		Name: image.Name,
+		// We must wait for the image to be ready, on followup reconciliation loops.
+		Ready: false,
+	})
+	return true, nil
+}
+
+// createCustomImage will create a new VPC Custom Image.
+func (s *VPCClusterScope) createCustomImage() (*vpcv1.Image, error) {
+	if s.IBMVPCCluster.Spec.Image == nil {
+		return nil, fmt.Errorf("error failed to create vpc custom image, no image spec defined")
+	}
+
+	// Collect the Resource Group ID.
+	var resourceGroupID *string
+	// Check Resource Group in Image spec.
+	if s.IBMVPCCluster.Spec.Image.ResourceGroup != nil {
+		if s.IBMVPCCluster.Spec.Image.ResourceGroup.ID != "" {
+			resourceGroupID = ptr.To(s.IBMVPCCluster.Spec.Image.ResourceGroup.ID)
+		} else if s.IBMVPCCluster.Spec.Image.ResourceGroup.Name != nil {
+			id, err := s.ResourceManagerClient.GetResourceGroupByName(*s.IBMVPCCluster.Spec.Image.ResourceGroup.Name)
+			if err != nil {
+				return nil, fmt.Errorf("error retrieving resource group by name: %w", err)
+			}
+			resourceGroupID = id.ID
+		}
+	} else {
+		// Otherwise, we will use the cluster Resource Group ID, as we expect to create all resources in that Resource Group.
+		id, err := s.GetResourceGroupID()
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving resource group id: %w", err)
+		}
+		resourceGroupID = ptr.To(id)
+	}
+
+	// We must have an OperatingSystem value supplied in order to create the Custom Image.
+	// NOTE(cjschaef): Perhaps we could try defaulting this value, so it isn't required for Custom Image creation.
+	if s.IBMVPCCluster.Spec.Image.OperatingSystem == nil {
```

Comment:
If you are planning to keep this, you can add this check into webhook.

Reply:
Sorry, I'm not sure what the webhook expected change would be, can you provide more details?

Reply:
Is this additional API changes, such as enhancing this function?
https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/5f99eed4ed831535c7f238a570d77494b5c98540/api/v1beta2/ibmvpccluster_webhook.go#L72

Reply:
Yes. But you can take that as an separate issue.

Reply:
I can do that until it gets removed in favor of webhook validation then.